build/Linux+BSD/portable/AppRun.in

```diff
@@ -30,6 +21,26 @@ case "$1" in
     "${APPDIR}/bin/portable-utils" "$@"
     ;;
   * )
+    # Workaround for Jack library
+    if [ "$1" = --dummy-jack ]; then
+       shift
+       export LD_LIBRARY_PATH="${APPDIR}/optional:${LD_LIBRARY_PATH}"
+    elif [ "$1" = --no-dummy-jack ]; then
+       shift
+    else
+       _PATH=$PATH
+       PATH=/usr/local/sbin:/sbin:$PATH
```

Comment:
Can you add a comment here? E.g.
```bash
PATH=/usr/local/sbin:/sbin:$PATH # these locations not in $PATH by default on Debian
```

It would also be good if all variable names were inside curly braces and quoted. E.g.

```bash
_PATH="${PATH}"
PATH="/usr/local/sbin:/sbin:${PATH}"
```

This disambiguates and avoids problems with variables that contain spaces. It may not be strictly necessary everywhere, but it is necessary in some places doing it everywhere increases consistency.

Reply:
I believe /sbin is not in PATH for non-root users on any (sensible) Linux distrbution?

Reply:
It does indeed. I will need to test myself because I am surprised that `ldconfig` is not in PATH by default, and if it genuinly is not, that this issue has not come up before.

Reply:
well, Jack not working with AppImages has come up quite frequently IIRC?

Reply:
Anyway, I think I've found a much better way:
```
LC_ALL=C ldd "${APPDIR}/bin/mscore@MSCORE_INSTALL_SUFFIX@" | grep "libjack.*not found"
```
Don't you think this is much more stable and elegant, matching the right architecture, as well?

By the way, I've noticed that the order I've arranged the script may break `$APPIMAGE_LIBRARY_PATH` advanced usage.

I've also located a -minor, I believe- [security issue](https://lwn.net/Articles/99372) with `$LD_LIBRARY_PATH`: if it contains an empty element (i.e. single colon, or colon at the start or end, or two consecutive colons) it makes the loader look into the current directory for libraries.

So:
```
export LD_LIBRARY_PATH="${APPDIR}/optional${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}"
```
and
```
export LD_LIBRARY_PATH="${APPIMAGE_LIBRARY_PATH:+${APPIMAGE_LIBRARY_PATH}:}${libs}:${libs}/qt5/lib${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}"
```
And `--dummy-jack` & `--no-dummy-jack` should be parsed anywhere in the command line (not just first argument).

I'm making some new commits.

___SPARE_COMMENTS___

Indeed, seems bizarre to find ldconfig in /usr/local/sbin but /usr/local is standard for local customization per [FHS](http://refspecs.linuxfoundation.org/FHS_3.0/fhs-3.0.html) and it doesn't cost anything to include it. Ok, it may break `grep`, or `echo` (?!), so I would rewrite it like this:
```
if PATH="/usr/local/sbin:/sbin:${PATH}" ldconfig -p \
| grep '^\s*libjack\.so\.0\s' >/dev/null 2>&1; then
...
```
which works as expected for `bash`.

Also, should include `/usr/sbin` ...

By the way I've found out that `ldconfig -p` isn't available on *BSD, but I suppose this is irrelevant as AppImages are not for *BSD. Right?

Reply:
Same goes for PATH, so better: `PATH=/usr/local/sbin:/sbin:${PATH-/bin}`  and  `export LD_LIBRARY_PATH="${APPDIR}/optional:${LD_LIBRARY_PATH-/lib}"`

Reply:
You're right about `PATH`, but I'm not going to use `PATH` manipulation, because I'm not going to use `ldconfig` at all! I'm going to use `ldd` on the MuseScore binary executable.

What's more, your syntax does not produce the desirable result:
```
export LD_LIBRARY_PATH="${APPDIR}/optional:${LD_LIBRARY_PATH-/lib}"
```
puts **one colon** anyway and then checks if `LD_LIBRARY_PATH` is set (**even if null**), in which case substitutes it's value (**even if null**). If unset (**not when null**) it substitutes `/lib` _preceding any system-wide configured library search path_! This is certainly not desirable!

Whereas:
```
export LD_LIBRARY_PATH="${APPDIR}/optional${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}"
```
checks if `LD_LIBRARY_PATH` is set (**and not null**) and -if so- substitutes **one colon** and the value of the variable (**certainly not null**). If unset (**or when null**) it substitutes _nothing_ (so, no spare colon).


I know, these `bash` syntax details are very complex and error prone! I always test them! And the colon here has multiple meanings...

Reply:
I haven't committed anything new yet because I'm trying to make the manpage in `mscore.1.in` and the help messages in `portable-utils.in` reflect my changes... Should I check anything else, regarding documentation, except for these two?

I've also found out that `install` & `uninstall` options for the AppImage are severely broken (the result is not functional, files are missing, others are misplaced), so I'm going to disable them with a message informing that they're not implemented, so as to save some users from polluting or even destroying their filesystem. Maybe it's simple to resolve, but I can't decide where to put the needed files and if MuseScore will finally pick them up.

Ok with disabling that? Should I open another pull request for it?

Reply:
I can confirm that `ldconfig` is in `/sbin` on Debian - strange that one never came up before!

`ldd` works in this particular case because the `mscore` binary is linked to `libjack.so.0`, but it wouldn't work for "dependencies of dependencies". This means it is not a general solution like the one my PR #5200 attempts to provide, but it is fine for locating Jack.

Good job spotting the flaw with empty components to LD_LIBRARY_PATH! The syntax:

```bash
VAR="<new_value>${VAR:+:${VAR}}"
```

is a good solution!

Reply:
> I can confirm that `ldconfig` is in `/sbin` on Debian - strange that one never came up before!

Most probably AppImage users don't use Jack for MuseScore and the relevant error message gets ignored as the thing just works for them.

> `ldd` works in this particular case because the `mscore` binary is linked to `libjack.so.0`, but it wouldn't work for "dependencies of dependencies". This means it is not a general solution like the one my PR #5200 attempts to provide, but it is fine for locating Jack.

Oh sorry, didn't know you're working on it!
Then why not run `ldd` on these dependencies as well?
(I've posted some comments on your PR)

Reply:
No worries! Your PR brought my attention to `ldconfig` being in `/sbin` on Debian.

`ldd` might not work for what I am trying to do in the other PR, so I will continue that discussion [over there](https://github.com/musescore/MuseScore/pull/5200#discussion_r307026696). There is no problem with you using `ldd` in this PR.